packages/block-library/src/video/edit-common-settings.js

```diff
@@ -50,28 +50,28 @@ const VideoSettings = ( { setAttributes, attributes } ) => {
 			<ToggleControl
 				label={ __( 'Autoplay' ) }
 				onChange={ toggleFactory.autoplay }
-				checked={ autoplay }
+				checked={ !! autoplay }
 				help={ getAutoplayHelp }
 			/>
 			<ToggleControl
 				label={ __( 'Loop' ) }
 				onChange={ toggleFactory.loop }
-				checked={ loop }
+				checked={ !! loop }
 			/>
 			<ToggleControl
 				label={ __( 'Muted' ) }
 				onChange={ toggleFactory.muted }
-				checked={ muted }
+				checked={ !! muted }
 			/>
 			<ToggleControl
 				label={ __( 'Playback controls' ) }
 				onChange={ toggleFactory.controls }
-				checked={ controls }
+				checked={ !! controls }
```

Comment:
Strictly speaking, this is not necessary since [`control` has a default value](https://github.com/WordPress/gutenberg/blob/bae8ee104f2ee5c10ad591af15d98f2df5290c31/packages/block-library/src/video/block.json#L28), but I handled this to unify with the other controls.